src/net/dash/manifest/helpers.ts

```diff
@@ -237,7 +238,7 @@ function inferAdaptationType(adaptation: IAdaptationDash) : string {
     if (role) {
       if (
         role.schemeIdUri === "urn:mpeg:dash:role:2011" &&
-        role.value === "subtitle"
+        SUPPORTED_TEXT_TYPE.indexOf(role.value.toString()) >= 0
```

Comment:
arrayIncludes from ``utils`` would be more lisible.

Also, is there a difference between "subtitle" and "caption"? Is "caption" obligatory for the hard-of-hearing or is there a subtility between them?